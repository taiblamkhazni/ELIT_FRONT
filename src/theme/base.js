/**
 * @file base.js
 * @brief Ce fichier d√©finit le composants de style base.
 */
export const base = {
    breakpoints: ["768px"],
    font: "'Roboto sans', sans-serif",
    backgrounds: {
        light: "#E5E5E5",
        gris: "#F9F9F9",
        white: "#FFFFFF",
    },
    fontWeights: {
        light: "300",
        regular: "400",
        bold: "700",
        black: "900",
    },
    fontSizes: {
        Kilo: "32px",
        Hecto: "24px",
        Deca: "20px",
        Base: "18px",
        Deci: "16px",
        Centi: "14px",
    },
    lineHeights: {
        Kilo: "54px",
        Hecto: "48px",
        Deca: "40px",
        Base: "40px",
        Deci: "32px",
        Centi: "24px",
    },
    colors: {
        primaires: {
            blueLight: "#F7FBFF",
            blue: "#116E9C",
            blueDark: "#1F1A28",
            white: "#FFFFFF",
        },
        secondaires: {
            violet: "#7B83E2",
            vertBlue: "#39B7BF",
            blue: "#6FC1E2",
            orange: "#FFAC5F",
            vert: "#DFE997",
            rose: "#FA927D",
            jaune: "#FFFBA0",
            gris: "#F9F9F9",
            grisMedium: "#CCCCCC",
            grisLight: "#E9E9E9",
            grisDark: "#7A7A7A",
        },
        avertissements: {
            succes: "#10B581",
            danger: "#C91432",
        },
    },
}
