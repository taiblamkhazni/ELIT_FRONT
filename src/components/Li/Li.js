/**
 * @file Li.js
 * @brief This module exports Input component
 */
import styled from "styled-components"

/** Custom Li of a list Component */
export const NotifyLi = styled.li`
  border-bottom: 1px solid ${({ theme }) => theme.colors.secondaires.grisMedium};
  padding: 12px 0px 7px 0px;
  min-height: 56px;
`
