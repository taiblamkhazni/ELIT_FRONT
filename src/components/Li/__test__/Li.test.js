/**
 * @file Li.test.js
 * @brief Ce fichier contient des tests pour le composant Li.
 */
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import { base } from "theme/base.js"

import { NotifyLi } from "../Li"

import "jest-styled-components"

const themeColor = base.colors.secondaires.grisMedium

describe("<NotifyLi />", () => {
    test("Li has the correct styles", () => {
        const wrapper = renderer
            .create(
                <ThemeProvider theme={base}>
                    <NotifyLi />
                </ThemeProvider>)
            .toJSON()
        expect(wrapper).toMatchSnapshot()
        expect(wrapper).toHaveStyleRule("border-bottom", `1px solid ${themeColor}`)
        expect(wrapper).toHaveStyleRule("padding", "12px 0px 7px 0px")
        expect(wrapper).toHaveStyleRule("min-height", "56px")
    })
})
