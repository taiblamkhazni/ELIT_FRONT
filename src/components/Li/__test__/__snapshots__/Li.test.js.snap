// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotifyLi /> Li has the correct styles 1`] = `
.c0 {
  border-bottom: 1px solid #CCCCCC;
  padding: 12px 0px 7px 0px;
  min-height: 56px;
}

<li
  className="c0"
/>
`;
